stix_shifter_modules/darktrace/stix_translation/query_constructor.py

```diff
@@ -355,3 +371,238 @@ def translate_pattern(pattern: Pattern, data_model_mapping, options):
     translated_query_strings = QueryStringPatternTranslator(pattern, data_model_mapping, options)
     queries = translated_query_strings.qualified_queries
     return queries
+
+
+class QuerySeparator:
+    """
+    split the translated query based on query length
+    """
+
+    def split_query(self, query):
+        """
+        split the query based on query length limit.
+        Between two statements if OR is present it will split if AND is present it wont split.
+        param query: str
+        return queries: list
+        """
+        try:
+            queries = []
+            recursion_limit = sys.getrecursionlimit()
+            sys.setrecursionlimit(2000)
```

Comment:
why are you changing the system recursion limit? how did you set the limit to 2000?

Reply:
While converting the translated query to a nested list, we are using the nestedExpr function (imported from module pyparsing).
It uses recursive methods to achieve this. Standard Python modules using the default system recursion limit of 1000.
For the larger query, we got a recursive limitation error for the default limit of 1000. So we doubled the limit to 2000 after that issue was resolved.

Reply:
The recursion limit is actually brought into stix-shifter as an argument since the default limit was found to not be enough for long queries. https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter/scripts/stix_shifter.py#L319

Reply:
The translate command accepts a recursion limit as an argument, whereas the execute command does not. 
recursion_limit argument is not defined in the execute parser. 
Only the translate command allows us to set the recursion_limit. Hence we have updated to use recursion_limit in our code.

Reply:
Changing recursion limit arbitrarily may cause some other memory issue. What's the largest query you tried? Are you sure the 2000 is enough for even large queries? 

Yes, right now its an argument for translate command. I think you can still run test in CLI without the execute command. Just run the translate and transmission commands separately by passing all the required parameters. OR you can just add that argument in your workspace and try the execute command. https://github.com/opencybersecurityalliance/stix-shifter/blob/689c5aff2346fbb36447f58f969ab16f0f2efa66/stix_shifter/scripts/stix_shifter.py#L71

Reply:
1. Below is the STIX pattern we tried, which has 100 comparison expressions.
"[network-traffic:dst_port=1001 OR network-traffic:dst_port=1002 OR network-traffic:dst_port=1003 OR network-traffic:dst_port=1004 OR network-traffic:dst_port=1005 OR network-traffic:dst_port=1006 OR network-traffic:dst_port=1007 OR network-traffic:dst_port=1008 OR network-traffic:dst_port=1009 OR network-traffic:dst_port=1010 OR network-traffic:dst_port=1011 OR network-traffic:dst_port=1012 OR network-traffic:dst_port=1013 OR network-traffic:dst_port=1014 OR network-traffic:dst_port=1015 OR network-traffic:dst_port=1016 OR network-traffic:dst_port=1017 OR network-traffic:dst_port=1018 OR network-traffic:dst_port=1019 OR network-traffic:dst_port=1020 OR network-traffic:dst_port=1021 OR network-traffic:dst_port=1022 OR network-traffic:dst_port=1023 OR network-traffic:dst_port=1024 OR network-traffic:dst_port=1025 OR network-traffic:dst_port=1026 OR network-traffic:dst_port=1027 OR network-traffic:dst_port=1028 OR network-traffic:dst_port=1029 OR network-traffic:dst_port=1030 OR network-traffic:dst_port=1031 OR network-traffic:dst_port=1032 OR network-traffic:dst_port=1033 OR network-traffic:dst_port=1034 OR network-traffic:dst_port=1035 OR network-traffic:dst_port=1036 OR network-traffic:dst_port=1037 OR network-traffic:dst_port=1038 OR network-traffic:dst_port=1039 OR network-traffic:dst_port=1040 OR network-traffic:dst_port=1041 OR network-traffic:dst_port=1042 OR network-traffic:dst_port=1043 OR network-traffic:dst_port=1044 OR network-traffic:dst_port=1045 OR network-traffic:dst_port=1046 OR network-traffic:dst_port=1047 OR network-traffic:dst_port=1048 OR network-traffic:dst_port=1049 OR network-traffic:dst_port=1050 OR network-traffic:dst_port=1051 OR network-traffic:dst_port=1052 OR network-traffic:dst_port=1053 OR network-traffic:dst_port=1054 OR network-traffic:dst_port=1055 OR network-traffic:dst_port=1056 OR network-traffic:dst_port=1057 OR network-traffic:dst_port=1058 OR network-traffic:dst_port=1059 OR network-traffic:dst_port=1060 OR network-traffic:dst_port=1061 OR network-traffic:dst_port=1062 OR network-traffic:dst_port=1063 OR network-traffic:dst_port=1064 OR network-traffic:dst_port=1065 OR network-traffic:dst_port=1066 OR network-traffic:dst_port=1067 OR network-traffic:dst_port=1068 OR network-traffic:dst_port=1069 OR network-traffic:dst_port=1070 OR network-traffic:dst_port=1071 OR network-traffic:dst_port=1072 OR network-traffic:dst_port=1073 OR network-traffic:dst_port=1074 OR network-traffic:dst_port=1075 OR network-traffic:dst_port=1076 OR network-traffic:dst_port=1077 OR network-traffic:dst_port=1078 OR network-traffic:dst_port=1079 OR network-traffic:dst_port=1080 OR network-traffic:dst_port=1081 OR network-traffic:dst_port=1082 OR network-traffic:dst_port=1083 OR network-traffic:dst_port=1084 OR network-traffic:dst_port=1085 OR network-traffic:dst_port=1086 OR network-traffic:dst_port=1087 OR network-traffic:dst_port=1088 OR network-traffic:dst_port=1089 OR network-traffic:dst_port=1090 OR network-traffic:dst_port=1091 OR network-traffic:dst_port=1092 OR network-traffic:dst_port=1093 OR network-traffic:dst_port=1094 OR network-traffic:dst_port=1095 OR network-traffic:dst_port=1096 OR network-traffic:dst_port=1097 OR network-traffic:dst_port=1098 OR network-traffic:dst_port=1099 OR network-traffic:dst_port=1100] START t'2022-11-01T00:00:00.000Z' STOP t'2022-12-19T11:00:00.003Z'"


2. We tried translate command by passing the recursion limit as argument and the transmission command separately, and it worked. so removed the code to set the recursion limit. We have tested the query from CP4S also and it works fine.